dbt/include/snowflake/macros/materializations/incremental.sql

```diff
@@ -20,11 +20,19 @@
        Otherwise, play it safe by using a temporary table.
   #} */
 
+  {% if language == "python" and tmp_relation_type == "table" %}
+    {% do exceptions.raise_compiler_error("Python models currently only support temp tables. ") %}
+  {% endif %}
+
+  {% if strategy == "delete+insert" and tmp_relation_type == "table" %}
+    {% do exceptions.raise_compiler_error("Possiblity of inconsistant results when using temp table with delete+insert") %}
```

Comment:
Could we explicitly tell the user what they configured versus what the allowed value is? We'll also want to fix the spellings of "possibility" and "inconsistent" (if we keep both of those in the final message).

Something like the following (which I didn't test out)?

```suggestion
    {% do exceptions.raise_compiler_error(f"Python models only support 'table' for 'tmp_relation_type', but '{tmp_relation_type}' was specified. ") %}
  {% endif %}

  {% if strategy == "delete+insert" and unique_key is not none and tmp_relation_type != "table" %}
    {% do exceptions.raise_compiler_error(f"In order to guarantee consistent results when 'unique_key' is given, the 'delete+insert' strategy only supports 'table' for 'tmp_relation_type'  but '{tmp_relation_type}' was specified.") %}
```

Alternatively, we can say something very simple like "'tmp_relation_type' is not allowed for Python models".

Reply:
updated messaging, did some local testing will. continue in a bit looking good thus far